src/main/java/mate.academy/lib/Injector.java

```diff
@@ -55,11 +54,11 @@ private Object newInstanceOf(Class<?> clazz) {
         }
     }
 
-    private Class<?> findImplementationOf(Class<?> interfaceClazz) {
-        Map<Class<?>, Class<?>> implementationsOfInterfaces = new HashMap<>();
-        implementationsOfInterfaces.put(FileReaderService.class, FileReaderServiceImpl.class);
-        implementationsOfInterfaces.put(ProductParser.class, ProductParserImpl.class);
-        implementationsOfInterfaces.put(ProductService.class, ProductServiceImpl.class);
-        return implementationsOfInterfaces.get(interfaceClazz);
+    private Class<?> getImplementation(Class<?> interfaceClazz) {
+        Map<Class<?>, Class<?>> implementations = new HashMap<>();
+        implementations.put(FileReaderService.class, FileReaderServiceImpl.class);
+        implementations.put(ProductParser.class, ProductParserImpl.class);
+        implementations.put(ProductService.class, ProductServiceImpl.class);
+        return implementations.get(interfaceClazz);
```

Comment:
again - if the class that is passed to this method is not of interface type - should we look though the map?